src/main/java/org/jenkinsci/plugins/ewm/steps/model/ExternalWorkspace.java

```diff
@@ -52,4 +63,21 @@ public String getMasterMountPoint() {
     public String getPathOnDisk() {
         return pathOnDisk;
     }
+
+    @Override
+    public String getDisplayName() {
+        return String.format("External Workspace on Disk ID: %s from Disk Pool ID: %s", diskId, diskPoolId);
+    }
+
+    @Restricted(NoExternalUse.class)
+    // TODO how to reference this from index.jelly file?
+    public DirectoryBrowserSupport doWs(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, InterruptedException {
```

Comment:
You will need a doDynamic method() in the action, which will redirect requests like RUN_URL/externalWorkspace/id to this model object